server/src/main/java/org/elasticsearch/common/filesystem/FileSystemNatives.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.filesystem;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.util.Constants;
+
+import java.nio.file.Path;
+import java.util.OptionalLong;
+
+/**
+ * This class provides utility methods for calling some native methods related to filesystems.
+ */
+public final class FileSystemNatives {
+
+    private static final Logger logger = LogManager.getLogger(FileSystemNatives.class);
+
+    interface Provider {
+        OptionalLong allocatedSizeInBytes(Path path);
+    }
+
+    private static final class NoopFileSystemNativesProvider implements Provider {
```

Comment:
I was wondering if could annotate `Provider` as `@FunctionInterface` and implement `NoopFileSystemNativesProvider` as `Provider NO_OP_FILE_SYSTEM_NATIVES_PROVIDER = path -> OptionalLong.empty()` or just use it as lambda without the constant alas `return path -> OptionalLong.empty()`

Reply:
Ok, I pushed e1aff78325491898c47512ebf27bf58eda5af6cb